lib/chef/knife/ec_restore.rb

```diff
@@ -238,11 +238,17 @@ def upload_org_data(name)
           # pivotal and members of the admins group. Use the same strategy
           # above here.
           #
-          ['admins', 'billing-admins', 'public_key_read_access'].each do |group|
+          groups = ['admins', 'billing-admins']
+          groups.push('public_key_read_access') if server.supports_public_key_read_access?
+
+          groups.each do |group|
             restore_group(chef_fs_config, group, :clients => false)
           end
 
-          ['/acls/groups/billing-admins.json', '/acls/groups/public_key_read_access.json'].each do |acl|
+          acls_groups_paths = ['/acls/groups/billing-admins.json']
+          acls_groups_paths.push('/acls/groups/public_key_read_access.json') if server.supports_public_key_read_access?
```

Comment:
🤔 the 3 `put_acl` HTTP 400 errors are actually coming from the restoring user acls section and it appears to only be related to the `read` permission:

```
Restoring user ACLs
PUTTING users/admin/_acl/create {"actors"=>["admin", "pivotal"], "groups"=>[]}
PUTTING users/admin/_acl/read {"actors"=>["admin", "pivotal"], "groups"=>["brewinc_global_admins"]}
PUTTING users/delivery/_acl/create {"actors"=>["delivery", "pivotal"], "groups"=>[]}
PUTTING users/delivery/_acl/read {"actors"=>["delivery", "pivotal"], "groups"=>["brewinc_global_admins"]}
PUTTING users/john/_acl/create {"actors"=>["john", "pivotal"], "groups"=>[]}
PUTTING users/john/_acl/read {"actors"=>["john", "pivotal"], "groups"=>["acmeinc_global_admins"]}
```

```
192.168.200.1 - - [26/Jul/2017:01:02:38 +0000]  "PUT /users/admin/_acl/read HTTP/1.1" 400 "0.013" 85 "-" "Chef Knife/13.2.20 (ruby-2.4.1-p111; ohai-13.2.0; x86_64-darwin14; +https://chef.io)" "127.0.0.1:8000" "400" "0.011" "13.2.20" "algorithm=sha1;version=1.1;" "pivotal" "2017-07-26T01:02:38Z" "JdmKCWPL/xK0R92iOrjVfmJGVFo=" 1181
192.168.200.1 - - [26/Jul/2017:01:02:38 +0000]  "PUT /users/delivery/_acl/read HTTP/1.1" 400 "0.008" 85 "-" "Chef Knife/13.2.20 (ruby-2.4.1-p111; ohai-13.2.0; x86_64-darwin14; +https://chef.io)" "127.0.0.1:8000" "400" "0.007" "13.2.20" "algorithm=sha1;version=1.1;" "pivotal" "2017-07-26T01:02:38Z" "WVBLmVwVGSufar2rlHG7Qjnnk4s=" 1187
192.168.200.1 - - [26/Jul/2017:01:02:38 +0000]  "PUT /users/john/_acl/read HTTP/1.1" 400 "0.011" 85 "-" "Chef Knife/13.2.20 (ruby-2.4.1-p111; ohai-13.2.0; x86_64-darwin14; +https://chef.io)" "127.0.0.1:8000" "400" "0.008" "13.2.20" "algorithm=sha1;version=1.1;" "pivotal" "2017-07-26T01:02:38Z" "+PWx9k+Z+PaRwd3FXA3yRGBlINY=" 1179
```